### PR TITLE
Use a column for dragging and dropping

### PR DESCRIPTION
**Pull request by navneetrai**
Fixes: #60 

- Added a property named `sortUsing` which will be the field using which you can sort models.

### CODE REVIEW

**Comment by navneetrai** [comment]
I am still working on updating tests but apart from that this should be complete.

File: README.md [hunk 1]
```diff
@@ -67,6 +67,28 @@ You can also update, re-arrange (for one-to-many relationships), and delete rela
...
+ To rearrange a model you should specify an `integer` field which will be used to sort models. For example, in the following code we will use a field named `weight` to store the sorting order for `Images`.
+ 
+ ```php
+ namespace App\Nova;
+ 
+ use Laravel\Nova\Fields\Image;
+ 
+ class BlogPost extends Resource
+ {
+     //...
```
Comment (brandonferens) [inline README.md]: @navneetrai I don't think we need this extra class info here. The fields method is good enough

File: README.md [hunk 2]
```diff
@@ -67,6 +67,28 @@ You can also update, re-arrange (for one-to-many relationships), and delete rela
...
+     public function fields(Request $request)
+     {
+         return [
+             //...
+ 
+             HasMany::make('Images', 'images', Image::class)->inline()->sortUsing('weight'),
+         ];
+     }
+ }
```
Comment (brandonferens) [inline README.md]: Same with this closing bracket

File: resources/js/components/FormField.vue [hunk 3]
```diff
@@ -9,6 +9,7 @@
       <draggable
         v-model="items"
         handle=".relationship-item-handle"
+        :disabled="!field.sortable"
```
Comment (brandonferens) [inline resources/js/components/FormField.vue]: Add a space after "not"
```suggestion
        :disabled="! field.sortable"
```

File: resources/js/components/FormField.vue [hunk 4]
```diff
@@ -89,16 +90,16 @@ export default {
          */
         setInitialValue() {
             this.items = Array.isArray(this.field.value) ? this.field.value : [];
-            this.items = this.items.map(item=>{
-                return { 'id': this.getNextId(), 'fields':item }
+            this.items = this.items.map((item, index) => {
+                return { 'id': this.getNextId(), 'modelId': this.field.models[index], 'fields':item }
```
Comment (brandonferens) [inline resources/js/components/FormField.vue]: Let's put each key on its own line
```suggestion
                return { 
                    'id': this.getNextId(),
                    'modelId': this.field.models[index],
                    'fields':item
                };
```

File: resources/js/components/FormField.vue [hunk 5]
```diff
@@ -89,16 +90,16 @@ export default {
...
             });
 
             if(this.field.singular){
                 this.items.splice(1);
             }
 
             if(this.field.addChildAtStart && (this.items.length === 0)){
-                this.items.push({ 'id': this.getNextId(), 'fields': {...this.field.settings}});
+                this.items.push({ 'id': this.getNextId(), 'modelId': 0, 'fields': {...this.field.settings}});
```
Comment (brandonferens) [inline resources/js/components/FormField.vue]: Same with lines

File: resources/js/components/FormField.vue [hunk 6]
```diff
@@ -145,7 +146,7 @@ export default {
 
         addItem(){
             let value = [...this.items];
-            value.push({ 'id': this.getNextId(), 'fields': {...this.field.settings}});
+            value.push({ 'id': this.getNextId(), 'modelId': 0, 'fields': {...this.field.settings}});
```
Comment (brandonferens) [inline resources/js/components/FormField.vue]: And again with lines

File: resources/js/components/RelationshipFormItem.vue [hunk 7]
```diff
@@ -117,17 +117,18 @@
             },
 
             fill(formData, parentAttrib) {
+	            formData.append(`${parentAttrib}[${this.id}][modelId]`, this.modelId);
```
Comment (brandonferens) [inline resources/js/components/RelationshipFormItem.vue]: Needs some formatting
```suggestion
	          formData.append(`${parentAttrib}[${this.id}][modelId]`, this.modelId);
	          
```

File: resources/js/components/RelationshipFormItem.vue [hunk 8]
```diff
@@ -117,17 +117,18 @@
...
                         if (keyParts.length === 1) {
-                            formData.append(`${parentAttrib}[${this.id}][${key}]`, value);
+                            formData.append(`${parentAttrib}[${this.id}][values][${key}]`, value);
                         } else {
                             let parentParts = parentAttrib.split('_');
                             let attrib = keyParts.slice(parentParts.length + 1).join('_');
 
-                            formData.append(`${parentAttrib}[${this.id}][${attrib}]`, value);
+                            formData.append(`${parentAttrib}[${this.id}][values][${attrib}]`, value);
                         }
```
Comment (brandonferens) [inline resources/js/components/RelationshipFormItem.vue]: Remove the else if possible. You could return at line 126 and remove the else, and return at line 131 too.

File: src/Observers/HasManyObserver.php [hunk 9]
```diff
@@ -11,20 +12,21 @@ class HasManyObserver extends BaseObserver
...
-        }
+            if (empty($childModel)) {
+                $model->{$attribute}()->create($value[$i]['fields']);
 
-        if ($count < count($childModels)) {
-            for ($i = $count; $i < count($childModels); $i++) {
-                $childModels[$i]->delete();
+                continue;
             }
+            $childModel->update($value[$i]['fields']);
```
Comment (brandonferens) [inline src/Observers/HasManyObserver.php]: Add an empty line above

File: README.md [hunk 10]
```diff
@@ -67,6 +67,13 @@ You can also update, re-arrange (for one-to-many relationships), and delete rela
 
 ![Rearrange Models](https://raw.githubusercontent.com/kirschbaum-development/nova-inline-relationship/master/screenshots/UpdateView.png "Rearrange Models")
 
+ To rearrange a model you should specify an `integer` field which will be used to sort models. For example, in the following code we will use a field named `weight` to store the sorting order for `Images`.
+ 
+ ```php
+HasMany::make('Images', 'images', Image::class)->inline()->sortUsing('weight'),
+ ```
```
Comment (brandonferens) [inline README.md]:
```suggestion
If you need the ability to sort your models, you can leverage this functionality using the `sortUsing()` method. You must pass as the first parameter the name of the field to sort by. In the following code example we use a field named `weight` to store the sorting order for `Images`:
 
 ```php
HasMany::make('Images', 'images', Image::class)->inline()->sortUsing('weight'),
 ```
```

Comment (brandonferens) [inline README.md]:
```suggestion
You can add drag and drop functionality for related models to update their order by using the `sortUsing()` method. In the following code example we use a field named `order` to store the sort order for the `Images` model:
 
 ```php
HasMany::make('Images')->inline()->sortUsing('order'),
 ```
```

**Pull request merged**